src/util/type.sml

```diff
@@ -0,0 +1,152 @@
+import "util.logging";
+
```

Comment:
Consider adding a signature so that the functions are documented clearly for others to use.